### PR TITLE
Docs: Developer section

### PR DESCRIPTION
**Pull request by camiinthisthang**
this section links out to other documentation

### CODE REVIEW

**Comment by camiinthisthang** [comment]
ignore linter fails, fix is in another PR currently waiting for approval

File: src/SUMMARY.md
Comment (adlerjohn) [inline src/SUMMARY.md]: Maybe pedantic, but change to TypeScript (capital S) everywhere.

```suggestion
  - [TypeScript SDK](./for-developers/ts-sdk.md)
```

File: src/for-developers/indexer.md
Comment (adlerjohn) [inline src/for-developers/indexer.md]:
```suggestion
The Fuel Indexer is a standalone binary that can be used to index various components of the blockchain. These indexable components include blocks, transactions, and receipts and state within a Fuel network, allowing for high-performance read-only access to the blockchain for advanced dApp use-cases.
```

File: src/for-developers/indexer.md
Comment (adlerjohn) [inline src/for-developers/indexer.md]: Hmmm I was under the impression that the Rust indexing was being phased out. cc @ra0x3

File: src/for-developers/rust-sdk.md
Comment (adlerjohn) [inline src/for-developers/rust-sdk.md]: Compiling is actually not currently supported as an SDK feature (it used to be)!

```suggestion
- Deploying, and testing Sway contracts;
```

File: src/for-developers/rust-sdk.md
Comment (adlerjohn) [inline src/for-developers/rust-sdk.md]:
```suggestion
- [Rust SDK Docs](https://fuellabs.github.io/fuels-rs/latest/)
```

File: src/for-developers/sway.md
Comment (adlerjohn) [inline src/for-developers/sway.md]:
```suggestion
- [Official Sway docs](https://fuellabs.github.io/sway/latest/)
```

File: src/for-developers/sway.md
Comment (adlerjohn) [inline src/for-developers/sway.md]: In this case, there's very little chance we'll rename the examples directory, so a permalink isn't necessary.

```suggestion
- [Sway examples repo](https://github.com/FuelLabs/sway/tree/master/examples)
```

File: src/for-developers/ts-sdk.md
Comment (adlerjohn) [inline src/for-developers/ts-sdk.md]:
```suggestion
- [TS SDK Repo](https://github.com/FuelLabs/fuels-ts)
```

File: src/for-developers/wallet-faucet.md
Comment (adlerjohn) [inline src/for-developers/wallet-faucet.md]: This sentence is kind of weirdly placed? Kind of just floating there without continuity.

File: src/for-developers/wallet-faucet.md
Comment (adlerjohn) [inline src/for-developers/wallet-faucet.md]: If would be nice to actually use `console` code blocks to show the commands and their outputs here.

File: src/for-developers/indexer.md
Comment (ra0x3) [reply src/for-developers/indexer.md]: We can remove any references to native execution for now, as we're primarily concerned with WASM execution only.

**Comment by sezna** [comment]
> ignore linter fails, fix is in another PR currently waiting for approval

which PR is that?

**Comment by camiinthisthang** [comment]
> > ignore linter fails, fix is in another PR currently waiting for approval
> 
> 
> 
> which PR is that?
> 
> 

Sorry, I should've linked it here https://github.com/FuelLabs/fuel-docs/pull/21

File: src/for-developers/wallet-faucet.md
Comment (camiinthisthang) [reply src/for-developers/wallet-faucet.md]: decided to remove this section entirely and send readers to the official steps, so that we have one source of truth, but noted!

File: src/for-developers/indexer.md
Comment (ra0x3) [inline src/for-developers/indexer.md:3]: - Just an FYI I copied this blurb for the Indexer book, but looks like there's a small grammatical error with an extra `and` here. 
- Should maybe be `...blocks, transactions, receipts, and state within ...` ?

Thanks @deekerno for the eagle eye 👁️

Comment (camiinthisthang) [reply src/for-developers/indexer.md:3]: Nice catch! I copied that blurb from [the Indexer docs](https://fuellabs.github.io/fuel-indexer/master/the-fuel-indexer.html?highlight=standalone%20binar#fuel-indexer) so I will also make a PR to fix it there!

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**
